clio-util/src/main/scala/org/broadinstitute/clio/util/json/StateDecoder.scala

```diff
@@ -78,7 +78,7 @@ object StateDecoder extends LowPriorityStateDecoder {
 
     field.as[Option[A]] match {
       // Found the field. Delete it, and return the modified cursor.
-      case Right(a @ Some(_)) => Right((field.delete, a))
+      case Right(a) if field.focus.isDefined => Right((field.delete, a))
```

Comment:
When decoding objects like `{"field": null}`, doing `field.as[Option[_]]` will return `Right(None)`, but we still want to call `field.delete`.